components/domains/steps/checkoutCard.tsx

```diff
@@ -664,12 +669,76 @@ const CheckoutCard: FunctionComponent<CheckoutCardProps> = ({
   const onCurrencySwitch = (type: CurrencyType) => {
     if (type !== CurrencyType.ETH) setLoadingPrice(true);
     setDisplayedCurrency(type);
+    setReducedDuration(0);
+    setReducedDurationToken(null);
+    setBetterReducedDuration(0);
   };
 
   const onUpsellChoice = (enable: boolean) => {
     updateFormState({ isUpselled: enable });
   };
 
+  useEffect(() => {
+    const duration = formState.duration;
+    if (!invalidBalance || !priceInEth) {
+      setReducedDuration(0);
+      setReducedDurationToken(null);
+      setBetterReducedDuration(0);
+      return;
+    }
+    let solutionFoundWithCurrentToken = 0;
+    for (let newDuration = duration - 1; newDuration > 0; newDuration--) {
+      const newPriceInEth = getPriceForDuration(priceInEth, newDuration);
+      let newPrice = newPriceInEth;
+      if (displayedCurrency !== CurrencyType.ETH && quoteData)
+        newPrice = getDomainPriceAltcoin(quoteData.quote, newPriceInEth);
+      const balance = tokenBalances[displayedCurrency];
+      if (!balance) continue;
+      if (BigInt(balance) >= BigInt(newPrice)) {
+        setReducedDuration(newDuration);
+        solutionFoundWithCurrentToken = newDuration;
+        break;
+      }
+    }
+    // If we reach this point, the user doesn't have enough balance for any duration
```

Comment:
I think you can update this comment, because it's possible you found a solution before, but now you're checking in other currencies if we can find an even better solution